adam_preprocessing/utils.py

```diff
@@ -71,50 +80,68 @@ def get_stroke_data(
             num_obj = len(features["objects"])
             if num_obj == 0:
                 continue
+            num_objs.append(num_obj)
 
-            # Sometimes the object detection/stroke extraction process picks up on multiple objects.
-            # When that happens, we want to combine them into one mega-example.
-            # In this mega-example, each distinct object is treated as its own connected component
-            # in a larger graph.
-            #
             # First we'll grab the coordinates since that's the cleaner part.
-            # Concatenate together the coords for each connected component to get a single
-            # (n_strokes_overall, 2, n_control_points) coordinates array
-            coords = np.concatenate(
-                [
-                    np.asarray(object_["stroke_graph"]["strokes_normalized_coordinates"])
-                    for object_ in features["objects"]
-                ],
-                axis=0,
-            )
+            if multi_object:
+                for object_ in features["objects"]:
+                    coords = np.asarray(object_["stroke_graph"]["strokes_normalized_coordinates"])
+                    curriculum_coords.append(coords)
+
+                    adj = np.asarray(object_["stroke_graph"]["adjacency_matrix"])
+                    curriculum_adjs.append(adj)
+
+                if int_curriculum_labels:
+                    integer_label, _ = label_from_object_language_tuple(
+                        language_tuple, situation_num
+                    )
+                    curriculum_labels.append(integer_label)
+
+            else:
+                # Merge objects
+                for object_ in features["objects"]:
+                    coords = np.asarray(object_["stroke_graph"]["strokes_normalized_coordinates"])
+                    curriculum_coords.append(coords)
+                    adj = np.asarray(object_["stroke_graph"]["adjacency_matrix"])
+                    curriculum_adjs.append(adj)
+                    if int_curriculum_labels:
+                        integer_label, _ = label_from_object_language_tuple(
+                            language_tuple, situation_num
+                        )
+                        curriculum_labels.append(integer_label)
```

Comment:
I'm assuming this is WIP code since it's so close to the multi-object code. Just noting that the `int_curriculum_labels` should go outside the loop.

Reply:
Sorry, I missed this when you posted it 2 days ago. I believe I have fixed it in the meanwhile; there is only one set of `(coords, adj, label)`extracted per feature file when in single-object mode